typing/env.ml

```diff
@@ -1255,7 +1254,18 @@ let find_ident_module id env =
   | Mod_unbound _ -> raise Not_found
   | Mod_persistent ->
       match Ident.to_global id with
-      | Some global_name -> find_pers_mod ~allow_hidden:true global_name
+      | Some global_name ->
+          let allow_excess_args =
+            (* This may be a global that arose by substituting instance arguments into
+               an overapproximated instance name, so we have to allow it to have more
+               arguments than expected. *)
```

Comment:
An example would be helpful here. In general, throughout this PR there are functions that takes `Global_module.Name.t`, for which `alow_excess_args` doesn't make sense?

Reply:
They can indeed: if `foo.ml` says `module M = M` and we have to approximate `M` as `M{P}{Q}` in `foo.cmi`, then if someone needs the type of `Foo[P:Int][Q:Float]`, we'll do substitution on the approximation `M{P}{Q}` and come up with `M[P:Int][Q:Float]`. At that point we _still_ haven't necessarily loaded `m.cmi` so it's possible for one or both of `P` and `Q` to be excess arguments, and it's not in general possible to detect that fact before we call `Persistent_env.find` and such.

Reply:
Agreed that an example would be good, though, so I'll add one

Reply:
Thanks that makes sense.